### PR TITLE
add weight when sampling

### PR DESCRIPTION
**Pull request by trangle1302**

### CODE REVIEW

**Comment by oeway** [comment]
Could you clarify how does the sampling weight work?

**Comment by trangle1302** [comment]
random.choices(self.sample_pool, weights=w,k=1) will apply the weight list w (generated as 1,2,3,...n with n is sample pool length) on the sample pool. So the most recent item in the sample pool will have highest weight n, the first item have weight 1. 
This is the easiest weighing. We can also do more sophisticated weights generation etc

File: interactive_trainer.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import sys
+import time
 import random
 import shutil
 import logging
 import json
-import time
 import traceback
 import warnings
+from stat import S_ISREG, ST_CTIME, ST_MODE
```
Comment (oeway) [inline interactive_trainer.py:10]: Where did you use these?

**Comment by oeway** [comment]
> random.choices(self.sample_pool, weights=w,k=1) will apply the weight list w (generated as 1,2,3,...n with n is sample pool length) on the sample pool. So the most recent item in the sample pool will have highest weight n, the first item have weight 1.
> This is the easiest weighing. We can also do more sophisticated weights generation etc

So the weights is based on the order in the sample pool? In that case, let's assume we added some samples and restart the program, the samples in the sample pool will be related from the file. In order to restore the sample pool order, I think we will need to sort the sample pool by their modification time of the corresponding annotation files, correct? Could you sort add that?

Could you please add some comments in the source code to explain this?

**Pull request closed without merging**